File: strings/Longest word in a sentence/longest_word_in_a_sentence.py
def longest_word_length(input_string):
    #spliting the sentence into words(word definition given below)
    words = input_string.split(" ") 
    #initializing the length with 0
    length = 0 

    for i in words:
        if(length < len(i)): #check for maximum length word
            length = len(i)
    return length

"""
Problem: In a given sentence(string) find the longest word.
(Word is any continuous stream of characters without spaces in between).

Example:
input: "This is the longest sentence"
**Here 'sentence' is longest word with length 8**
output: 8
"""

#drive code
input_string = input()
print(f"The length of the longest word in the given sentence is {longest_word_length(input_string)}")

File: strings/Longest word in a sentence/longest_word_in_a_sentence.cpp
#include <iostream>
using namespace std;
//function to find longest word
int longest_word_length(string str) {
   int length = str.length(); //length of the given string
   
   //intializing with 0
   int temp_length = 0;
   int new_length = 0;

   for (int i = 0; i < length; i++) {
      if (str[i] != ' ') //check of spaces
         new_length++;
      else {
         //if word found store length in temp_length
         temp_length = max(temp_length, new_length);
         //re-initialize new_length with 0 to store new word length
         new_length = 0;
      }
   }
   return max(temp_length, new_length); //return maximum length
}


int main() {
   string s;
   getline(cin , s) ; //input string
   cout <<"Length of the longest word in given sentence is "<<longest_word_length(s);
   return 0;
}